che-incubator/chectl
# Improve Eclipse Che deploy / update flow

### What does this PR do?

This PR contains several changes:
- Now `chectl` is able to deploy embedded (the same as cli) version of Che and all (with some limitations) previous versions.
- Added `version` flag to `server:deploy` command, that downloads templates for older versions. For future versions, update of `chectl` is required (so far the limitation is UX not technical).
- Added warning message for each command if newer `chectl` is available. For `server:deploy` and `server:update` commands interactive prompt is added.
- Deleted outdated installer `minishift-addon`.

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/18022

### How to test this PR?
- Deploy Eclipse Che using different installers.
- Use `--version` flag with different installers.
- Deploy Che using external templates.
- Update Che using operator and OLM installers.
- Change chectl version to see update warnings.

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)


### Reviewers

Reviewers, please comment how you tested the PR when approving it.
